cli/running/lua/play.lua

```diff
@@ -0,0 +1,192 @@
+--
+-- Several useful links:
+-- https://porkmail.org/era/unix/arg-max.html
+-- https://unix.stackexchange.com/a/120842
+
+local log = require('log')
+local xlog = require('xlog')
+local json = require('json')
+local netbox = require('net.box')
+local argparse = require('internal.argparse').parse
+
+local positional_arguments = {}
+local keyword_arguments = {}
+
+local function find_in_list(id, list)
+    if type(list) == 'number' then
+        return id == list
+    end
+    for _, val in ipairs(list) do
+        if val == id then
+            return true
+        end
+    end
+    return false
+end
+
+local function filter_xlog(gen, param, state, opts, cb)
+    local from, to, spaces = opts.from, opts.to, opts.space
+    local show_system, replicas = opts['show-system'], opts.replica
+
+    for lsn, record in gen, param, state do
+        local sid = record.BODY and record.BODY.space_id
+        local rid = record.HEADER.replica_id
+        if replicas and #replicas == 1 and replicas[1] == rid and lsn >= to then
+            -- Stop, as we've finished reading tuple with lsn == to
+            -- and the next lsn's will be bigger.
+            break
+        elseif (lsn < from) or (lsn >= to) or
+           (not spaces and sid and sid < 512 and not show_system) or
+           (spaces and (sid == nil or not find_in_list(sid, spaces))) or
+           (replicas and not find_in_list(rid, replicas)) then
+            -- Pass this tuple, luacheck: ignore.
+        else
+            cb(record)
+        end
+    end
+end
+
+local function play()
+    local opts = keyword_arguments
+    local uri = table.remove(positional_arguments, 1)
+    if uri == nil then
+        log.error("Internal error: empty URI is provided")
+        os.exit(1)
+    end
+    local remote = netbox.new(uri)
+    if not remote:wait_connected() then
+        log.error("Fatal error: no connection to the host '%s'", uri)
+        os.exit(1)
+    end
+    for _, file in ipairs(positional_arguments) do
+        print(string.format("• Play is processing file '%s' •", file))
+        local gen, param, state = xlog.pairs(file)
+        filter_xlog(gen, param, state, opts, function(record)
+            local sid = record.BODY and record.BODY.space_id
+            if sid ~= nil then
+                local args, so = {}, remote.space[sid]
+                if so == nil then
+                    log.error("Fatal error: no space #%s, stopping work", sid)
+                    os.exit(1)
+                end
+                table.insert(args, so)
+                table.insert(args, record.BODY.key)
+                table.insert(args, record.BODY.tuple)
+                table.insert(args, record.BODY.operations)
+                so[record.HEADER.type:lower()](unpack(args))
+            end
+        end)
+        print(string.format("• Done with file '%s' •", file))
+    end
+    print("\n• Play result: completed successfully •")
+    remote:close()
+end
+
+local function populate_arguments()
```

Comment:
Same questions as for `cat`.

Reply:
Simplified, `populate_arguments()` has been removed